### DISCUSSION TITLE
Backtest Realism

### DISCUSSION THREAD

**Discussion by limx0** [nautechsystems/nautilus_trader#379] [Ideas]
Following on from https://github.com/nautechsystems/nautilus_trader/issues/286, moving to a discussion.

I'd just like to jot down some medium term ideas / goals for improve the realism of nautilus backtests.
- Improved quoting logic (probabilistic fills based on a full order book)
- Maintaining order position in the orderbook (if someone is on the orderbook level, then your strategy joins, then someone joins behind you, we should simulate getting a fill if the level trades (as far as our order))
- An (optional) latency added to insert/delete/updates that a strategy sends
- Latency for strategy calculations (if a strategy goes off and computes something for 3s, we should walk the backtest clock forward by that amount of time before resuming).

If anyone has any further ideas or comments would be glad to hear them (or if anyone would like a hand at trying to take on one of these tasks, let me know!)

**Comment by crazy25000** [comment]
> Improved quoting logic (probabilistic fills based on a full order book)

I've read this in a few trading libraries, is this a successful/proven method? 

> Latency for strategy calculations (if a strategy goes off and computes something for 3s, we should walk the backtest clock forward by that amount of time before resuming).

+100 - I was thinking this exact thing!

**Reply by limx0** [reply]
> I've read this in a few trading libraries, is this a successful/proven method?

Not necessarily a proven method, but it might tell you how sensitive your strategy is to specific trades/fills given repeated runs of the same parameter set

**Reply by crazy25000** [reply]
Ohh that would be valuable to know.
